pkg/controllers/gateway/gateway_controller.go

```diff
@@ -147,7 +147,14 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// If the GatewayClass parameters are invalid, update the status and stop reconciling
 	params, err := getParams(ctx, r.Client, string(upstreamGateway.Spec.GatewayClassName))
 	if err != nil && IsInvalidParamsError(err) {
-		conditions.SetCondition(upstreamGateway.Status.Conditions, previous.Generation, string(gatewayv1beta1.GatewayConditionProgrammed), metav1.ConditionFalse, string(gatewayv1beta1.GatewayReasonPending), fmt.Sprintf("Invalid parameters in gateway class: %s", err.Error()))
+		programmedCondition := metav1.Condition{
+			Type:               string(gatewayv1beta1.GatewayConditionProgrammed),
+			Status:             metav1.ConditionFalse,
+			Reason:             string(gatewayv1beta1.GatewayReasonPending),
+			Message:            fmt.Sprintf("Invalid parameters in gateway class: %s", err.Error()),
+			ObservedGeneration: previous.Generation,
+		}
+		meta.SetStatusCondition(&upstreamGateway.Status.Conditions, programmedCondition)
```

Comment:
Would probably be better if we could just use `buildProgrammedCondition` from wherever we want to set the programmed condition and have it return the expected condition based on the error.